server/views.py

```diff
@@ -138,10 +125,10 @@ def cancel_upload(auth_token, sample_name, identifier):
     sample_name = str(sample_name)
     identifier = str(identifier)
 
-    if not get_file_data(identifier, 'upload_status'):
+    if not File.query.filter_by(identifier=identifier).first():
         return return_message('Error: Identifier does not exist', 404)
 
-    if get_file_data(identifier, 'upload_status') != 'complete':
+    if File.query.filter_by(identifier=identifier).first().upload_status != 'complete':
```

Comment:
Don't repeat the same query 4 times. Do it once, store the result, and then use the result.